static/js/composebox_typeahead.js

```diff
@@ -213,6 +213,11 @@ exports.tokenize_compose_str = function (s) {
         i -= 1;
         switch (s[i]) {
             case '#':
+                // If a '#' is immediately followed by a space, cancel the typeahead
+                if (s.charAt(i+1) === " ") {
+                    return '';
+                }
+                // falls through
```

Comment:
Ah, I guess I scrolled right past `autocomplete_checks`. That does seem like a better place for this.